pyomo/contrib/pynumero/interfaces/tests/test_pyomo_grey_box_nlp.py

```diff
@@ -0,0 +1,1773 @@
+        m.ipopt_zL_out = pyo.Suffix(direction=pyo.Suffix.IMPORT_EXPORT)
+        m.ipopt_zU_out = pyo.Suffix(direction=pyo.Suffix.IMPORT_EXPORT)
+
+        solver = pyo.SolverFactory('cyipopt')
+        status = solver.solve(m, tee=False)
+
+        self.assertAlmostEqual(pyo.value(m.p), 10.0, places=3)
+        self.assertAlmostEqual(pyo.value(m.egb.inputs['u']), 4.0, places=3)
+        self.assertAlmostEqual(pyo.value(m.egb.outputs['o']), 20.0, places=3)
+        self.assertAlmostEqual(pyo.value(m.dual[m.con]), 50.0, places=3)
+        self.assertAlmostEqual(m.dual[m.egb]['egb.output_constraints[o]'], -100.0, places=3)
+        self.assertAlmostEqual(pyo.value(m.ipopt_zL_out[m.egb.inputs['u']]), 500.0, places=3)
+        self.assertAlmostEqual(pyo.value(m.ipopt_zU_out[m.egb.inputs['u']]), 0.0, places=3)
+
+        del m.obj
+        m.obj = pyo.Objective(expr=-10*m.p**2)
+        status = solver.solve(m, tee=False)
+
+        self.assertAlmostEqual(pyo.value(m.p), 25.0, places=3)
+        self.assertAlmostEqual(pyo.value(m.egb.inputs['u']), 10.0, places=3)
+        self.assertAlmostEqual(pyo.value(m.egb.outputs['o']), 50.0, places=3)
+        self.assertAlmostEqual(pyo.value(m.dual[m.con]), -125.0, places=3)
+        self.assertAlmostEqual(m.dual[m.egb]['egb.output_constraints[o]'], 250.0, places=3)
+        self.assertAlmostEqual(pyo.value(m.ipopt_zL_out[m.egb.inputs['u']]), 0.0, places=3)
+        self.assertAlmostEqual(pyo.value(m.ipopt_zU_out[m.egb.inputs['u']]), -1250.0, places=3)
+
+        m = pyo.ConcreteModel()
+        m.p = pyo.Var(initialize=1)
+        m.egb = ExternalGreyBoxBlock()
+        m.egb.set_external_model(ex_models.OneOutputOneEquality())
+        m.con = pyo.Constraint(expr=4*m.p-2*m.egb.outputs['o'] == 0)
+        m.obj = pyo.Objective(expr=10*m.p**2)
+
+        # we want to check dual information so we need the suffixes
+        m.dual = pyo.Suffix(direction=pyo.Suffix.IMPORT_EXPORT)
+
+        solver = pyo.SolverFactory('cyipopt')
+        status = solver.solve(m, tee=False)
+
+        self.assertAlmostEqual(pyo.value(m.p), 2.5, places=3)
+        self.assertAlmostEqual(pyo.value(m.egb.inputs['u']), 1.0, places=3)
+        self.assertAlmostEqual(pyo.value(m.egb.outputs['o']), 5.0, places=3)
+        self.assertAlmostEqual(pyo.value(m.dual[m.con]), 12.5, places=3)
+        self.assertAlmostEqual(m.dual[m.egb]['egb.output_constraints[o]'], -25.0, places=3)
+        self.assertAlmostEqual(m.dual[m.egb]['egb.u2_con'], 62.5, places=3)
+
+if __name__ == '__main__':
+    TestExternalGreyBoxAsNLP().test_pressure_drop_single_equality()
+    TestExternalGreyBoxAsNLP().test_pressure_drop_two_outputs()
+    TestExternalGreyBoxAsNLP().test_pressure_drop_two_equalities()
+    TestExternalGreyBoxAsNLP().test_pressure_drop_two_equalities_two_outputs()
+    TestExternalGreyBoxAsNLP().test_scaling_all_missing()
+    TestExternalGreyBoxAsNLP().test_scaling_pyomo_model_only()
+    TestExternalGreyBoxAsNLP().test_scaling_greybox_only()
```

Comment:
Why are you calling out specific test cases and not just using
```python
if __name__ == '__main__':
    unittest.main()
```

Reply:
Because, I use these to debug particular tests when I don't want unittest capturing the text.